src/app/loadout-builder/utils.ts

```diff
@@ -74,3 +74,26 @@ export function getPower(items: DimItem[] | ProcessItem[]) {
 
   return Math.floor(power / numPoweredItems);
 }
+
+/** Gets the max energy allowed from the passed in UpgradeSpendTier */
+export function upgradeSpendTierToMaxEnergy(tier: UpgradeSpendTier, item: DimItem) {
+  const available = item.energy?.energyCapacity;
+  const isExotic = Boolean(item.equippingLabel);
+  if (available === undefined) {
+    return 0;
+  }
+
+  switch (tier) {
+    case UpgradeSpendTier.LegendaryShards:
+      return Math.max(7, available);
```

Comment:
Bonus points for calculating these limits from the socket's plug sequence: https://github.com/DestinyItemManager/DIM/blob/c3c3a979520aa5f8037fb0b342cf63469633b023/src/app/inventory/store/energy.ts#L20

Reply:
This one is done